### PR TITLE
Replace the type of client_id and actor_id in protobuf

### LINKED ISSUE #36: Replace ActorID type with UUID or Long instead of String to reduce the size of CRDT meta
The sizes of the types:

| Name        | Binary Size | String Size    | Features
|-------------|-------------|----------------|----------------
| [UUID]      | 16 bytes    | 36 chars       | configuration free, not sortable
| [shortuuid] | 16 bytes    | 22 chars       | configuration free, not sortable
| [Snowflake] | 8 bytes     | up to 20 chars | needs machine/DC configuration, needs central server, sortable
| [ObjectID]   | 12 bytes    | 24 chars  `<-- currently we use`      | configuration free, sortable
| xid         | 12 bytes    | 20 chars       | configuration free, sortable
| uint64         | 8 bytes    | N/A       | configuration free, needs central server, sortable

### PR DESCRIPTION
**Pull request by autumn-n**
**What this PR does / why we need it**:

> A 24 bytes hex-encoded string can be replaced with a 12 bytes of array to reduce size of marshalled data with Protobuf.

**Which issue(s) this PR fixes**:

> Fixes #36 

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE, It just reduce the size of data between client and server.
```

**Additional documentation**:

```docs

```

**Checklist**:
- [x] Added relevant tests or not required
- [x] Didn't break anything

### CODE REVIEW

**Comment by hackerwins** [comment]
@autumn-n Thank you for sending PR. We'll review this soon.

**Review by dc7303 — CHANGES_REQUESTED** [review]
Thanks for the wonderful work :+1:
I left some amendments and questions. After checking, please be honest with your opinion.

## Question
I think the modify the JS SDK also needs to be done to profile this PR. If you let me know if you're modifying the JS SDK, I think we'll see how we can help.
Are you modifying the JS SDK?

## About test
How about adding a unit test for the added method?
I think it would be nice to add the normal case and exception cases of the method. This is because it will serve as a document and a safeguard for developers who analyze and modify this method in the future.

File: pkg/document/time/actor_id.go [hunk 1]
```diff
@@ -79,6 +90,16 @@ func (id *ActorID) String() string {
 	return hex.EncodeToString(id[:])
 }
 
+// Bytes returns the bytes of ActorID itself.
+// If the receiver is nil, it would return empty array of byte.
+func (id *ActorID) Bytes() []byte {
+	if id == nil {
+		return []byte{}
```
[the remark this answers is not in the document]
Comment (autumn-n) [reply pkg/document/time/actor_id.go:97]: NP, I will follow it. Thanks for letting me know.

**Comment by autumn-n** [comment]
@dc7303 Thanks for your comments

> ## Question
> I think the modify the JS SDK also needs to be done to profile this PR. If you let me know if you're modifying the JS SDK, I think we'll see how we can help.
> Are you modifying the JS SDK?

Meanwhile, I'm already done. Once review has done, I will send a PR of JS SDK too.

> ## About test
> How about adding a unit test for the added method?
> I think it would be nice to add the normal case and exception cases of the method. This is because it will serve as a document and a safeguard for developers who analyze and modify this method in the future.

Yes, you are right. I should have done that before. I'm gonna do it. In addition, It would be also great if there were more tests to figure out the intentions for new contributors.

**Review by dc7303 — APPROVED** [review]
LGTM 👍 😄

**Review by hackerwins — APPROVED** [review]
LGTM. Thanks for the contribution.

**Pull request merged**